src/dynamics/video_player/player/player.js

```diff
@@ -3296,7 +3334,53 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
                         if (this.get("forciblymuted")) this.set("forciblymuted", false);
                     }.bind(this), 1);
                     this.set('clearDebounce', clearDebounce);
+                },
 
+                _recordPerformance: function(name) {
+                    const recordName = `${this.get(`performanceprefix`)}-${name}`;
+                    // Entry Types starting Chrome 59: "element", "event", "first-input",
+                    // "largest-contentful-paint", "layout-shift", "longtask", "mark",
+                    // "measure", "navigation", "paint", "resource"
+                    if (this.__previousMark) {
+                        performance.measure(recordName, this.__previousMark);
+                    }
+                    this.__previousMark = recordName;
+                    performance.mark(recordName);
+                },
+
+                __performanceObserverFunc: function(list) {
+                    list.getEntries().forEach((entry) => {
+                        let name;
+                        switch (entry.entryType) {
+                            case `navigation`:
+                                name = `${this.get(`performanceprefix`)}-navigation`;
+                                break;
+                            case `element`:
+                                name = entry.identifier || entry.name;
+                                break;
+                            default:
+                                name = entry.name
+                        }
+                        const index = this.get(`performancerecords`).findIndex(
+                            pr => (pr && pr.name === name)
+                        );
+                        const perf = index > -1 ? this.get(`performancerecords`)[index] : {
+                            name
+                        };
+                        const records = this.get(`performancerecordeditems`);
+                        records.forEach((k) => {
+                            if (!k.keys || !k.type || entry.entryType !== k.type) return;
+                            k.keys.forEach((key) => {
+                                if (entry[key]) perf[key] = entry[key];
+                            });
+                        });
+                        // to be able to trigger change event
+                        if (index > -1) this.get(`performancerecords`)[index] = perf;
+                        this.set(`performancerecords`, index > -1 ? this.get(`performancerecords`) : [
+                            ...this.get(`performancerecords`),
+                            perf
+                        ]);
+                    }, this);
```

Comment:
Hi @sambua, we should be able to unit test this

Reply:
Hi @ntshcalleia as js-dom not fully supporting performance API added headless integration test